src/libraries/System.Private.Uri/tests/ExtendedFunctionalTests/UriRelativeResolutionTest.cs

```diff
@@ -479,11 +478,11 @@ public void UriRelative_AbsoluteToAbsolute_CustomPortCarriedOver()
             Uri baseUri = new Uri("http://nothing.com/");
             Uri testUri = new Uri("https://specialPort.com:00065535/path?query#fragment");
 
-            int throwAway = testUri.Port; // Trigger parsing.
+            _ = testUri.Port; // Trigger parsing.
 
             Uri resultUri = new Uri(baseUri, testUri);
 
-            throwAway = resultUri.Port; // For Debugging.
+            _ = resultUri.Port; // For Debugging.
```

Comment:
What does the `// For Debugging` comment mean?  Is this functionally necessary?  It sounds like it is for the port one above.

Reply:
The code pattern above was probably used because there is a side-affect of doing a 'get' on the .Port property.

Will the behavior be the same if you use "_"?

Reply:
As far as I am aware, the behavior is the same as assigning to a dummy local, but it makes it obvious which values aren't used and avoids analyzer warnings

Reply:
@MihaZupan, I'm still not clear on what the "For Debugging" comment is referring to?

Reply:
Thinking about it more, I think this is a possible reason why it was used:

Uri is currently parsed in 3 phases (lazy):

1. In the ctor - identify the type of Uri, check the authority and other basic things to determine if the Uri is valid
2. Determine offsets for UserInfo, Host, Path, parse the port value
3. Determine the rest of offsets (Query, Fragment) and check if parts are in canonical form

This can make debugging harder. For example when debugging in Visual Studio, you must be careful to explicitly turn off the feature showing Locals - you can be stepping through the ctor and VS tries to evaluate properties like Query which trigger full parsing (before the initial ctor parsing even started), thus corrupting the Uri state under the debugger.

If the original author was debugging this code, they could have used code like
```c#
_ = uri.Port
Debugger.Launch();
```

to force more parsing before attaching the debugger, but forgot to remove it afterwards.

I've removed it.